apps/print/backend/src/util/ballot_styles.ts

```diff
@@ -75,3 +85,53 @@ export function findBallotStyleId(
     }
   }
 }
+
+export function addBallotsPropsToPrintCountRow(
+  election: Election,
+  printCountRow: BallotPrintCountRow
+): BallotPrintCount {
+  const { ballotStyleId } = printCountRow;
+  const ballotStyle = election.ballotStyles.find(
+    (bs) => bs.id === ballotStyleId
+  );
+  assert(ballotStyle, `No ballot style found with id ${ballotStyleId}`);
+
+  const precinct = assertDefined(
+    find(election.precincts, (p) => p.id === printCountRow.precinctId)
+  );
+  const precinctHasSplits = hasSplits(precinct);
+
+  const precinctAndSplitsForBallotStyle = getPrecinctsAndSplitsForBallotStyle({
+    election,
+    ballotStyle,
+  });
+
+  const matchingPrecinctOrSplit = assertDefined(
+    precinctAndSplitsForBallotStyle.find(
+      (ps) =>
+        ps.precinct.id === precinct.id &&
+        (precinctHasSplits ? ps.split !== undefined : ps.split === undefined)
```

Comment:
I could use some help sanity checking the logic for the Split case –

So first we get all precinctsAndSplits that use this ballot style. Then, if this precinctHasSplits, we find the precinct/Split that has a defined split, without needing to check the splitId (which I don't have). The assumption here is that only one split per precinct can have a specific ballot style, otherwise they would have the same districtIds which is not allowed by VxDesign ([link](https://github.com/votingworks/vxsuite/blob/main/apps/design/frontend/src/geography_screen.tsx#L621)). Does that sound right to you?